src/main/scala/org/broadinstitute/dsde/firecloud/HttpClient.scala

```diff
@@ -64,8 +64,11 @@ class HttpClient (requestContext: RequestContext) extends Actor
         log.debug("Got response: " + response)
         context.parent ! RequestComplete(response)
       case Failure(error) =>
-        log.error("External request failed", error)
-        context.parent ! RequestCompleteWithErrorReport(InternalServerError, "External request failed: " + error.getMessage, error)
+        val message = s"External request failed to ${externalRequest.uri.toString()} (${error.getMessage})"
+        val customException = new Exception(message, error)
```

Comment:
nitpick: always prefer `FireCloudException` instead of `Exception`. Other than that, PR is great, nice and simple!

( if all you change is `Exception` -> `FireCloudException`, I don't need to re-review )